pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTModifierNode.java

```diff
@@ -62,10 +64,16 @@ public boolean isTransient() {
         return (node.getModifiers().getJavaModifiers() & TRANSIENT) == TRANSIENT;
     }
 
+    // true if function has `@isTest` annotation or `testmethod` modifier
     public boolean isTest() {
         return node.getModifiers().isTest();
     }
 
+    // true if function has `testmethod` modifier
+    public boolean isTestMethod() {
```

Comment:
Now we have isTest and isTestMethod, and it's not clear from the name what the difference is. Please name this one something like `hasDeprecatedTestMethodModifier`. We should anyway prefer the other, so a cumbersome name is better for this one.

Also please use javadoc comments `/** Returns true if ... */` instead of end-of-line comments

Reply:
Understood. Makes a lot of sense to me - I thought it might get confusing which is why I added the comments but having the function named more clearly is of obvious benefit.

How does `has` work with the xpath syntax? `is` is clearly either ignored or is a recognised keyword or something to that effect. Just wondering how I would have to adjust the xpath rule to account for the changed method name.

Reply:
I renamed it to DeprecatedTestMethod as I couldn't work out the interaction of `has` prefix with xpath. IMO this also correlates better with the existing method names and the generated AST.

Reply:
Attributes prefixed with `has` are stripped in the same way as those starting with `is`. So the name would be `@DeprecatedTestMethodModifier`. I still think this would be preferable. `isDeprecatedTestMethod` sounds like `isTestMethod() && isDeprecated()`

Reply:
Thanks for clarifying this, done